### PR TITLE
fix: support nested repository identities end to end

### PR DESCRIPTION
**Pull request by ColeMurray**
## Summary

- add canonical shared helpers for formatting, parsing, and route-encoding repository identities
- preserve nested GitLab owner namespaces through web selection, settings, secrets, bot configuration, image builds, and control-plane routes
- resolve sandbox pull-request targets from the repository manifest and preserve canonical identity/path data
- document the nested-owner contract to prevent first-slash parsing regressions

## Root cause

PR #991 allowed slash-delimited repository owners in the sandbox repository model, but several downstream consumers still treated full names as exactly owner/name or inserted nested owners into URL paths without encoding them as one route segment. That truncated group/subgroup/project identities or produced routes with too many segments.

## Test plan

- npm test: shared, control-plane, web, and linear-bot (3,185 tests)
- control-plane integration-settings integration suite (33 tests)
- sandbox-runtime pytest suite (456 tests)
- modal-infra pytest suite (213 tests)
- typecheck: shared, control-plane, web, and linear-bot
- lint: shared, control-plane, web, and linear-bot
- Ruff checks for changed Python files
- production bundles: control-plane and linear-bot
- git diff --check

Follow-up to #991.

## Summary by CodeRabbit

* **New Features**
  * Added support for repositories with nested owners/namespaces across repository selection, session targets, integrations, image rebuild triggers, sandbox scope URLs, and SCM links.
  * Repository identity handling is now consistent, with canonical casing/normalization and safe encoding/decoding for API paths.
* **Bug Fixes**
  * Improved routing and request construction for repo-scoped operations, including stricter validation and clearer 400 errors for malformed path segments.
  * Normalized stored/returned routing-rule targets and per-repo override endpoints for mixed casing and nested paths.
* **Documentation**
  * Updated repository path/owner guidance to match the new canonical handling.
* **Tests**
  * Added/updated coverage for nested-owner parsing, encoding, and validation edge cases.

### CODE REVIEW

File: packages/control-plane/src/routes/shared.ts [hunk 1]
```diff
@@ -145,12 +145,16 @@ export async function parseJsonBody<T>(request: Request): Promise<T | Response>
...
-  const name = match.groups?.name;
-  if (!owner || !name) {
+  const encodedOwner = match.groups?.owner;
+  const encodedName = match.groups?.name;
+  if (!encodedOwner || !encodedName) {
     return error("Owner and name are required", 400);
   }
-  return { owner, name };
+  try {
+    return { owner: decodeURIComponent(encodedOwner), name: decodeURIComponent(encodedName) };
```
[the remark this answers is not in the document]
Comment (ColeMurray) [reply packages/control-plane/src/routes/shared.ts]: Done — addressed in `e30aca56`. I added `decodeRepositoryPathSegments` alongside the shared encoder and made `extractRepoParams` use it. It rejects decoded slashes in `repoName`, empty nested-owner segments, and malformed percent encoding, so all repository routes now receive the same canonical identity shape. Regression coverage was added for each invalid case.

**Comment by ColeMurray** [comment]
Done — replaced the literal subprocess timeout with `TOOL_SUBPROCESS_TIMEOUT_SECONDS`, preserving the existing behavior while following the repository’s duration naming convention.

**Comment by ColeMurray** [comment]
happy #1000!
